src/lib/browser.ts

```diff
@@ -52,28 +56,38 @@ export async function* generate<T>(
 						headers[tmp.shift().toLowerCase()] = tmp.join(': ');
 					}
 
-					let body = current.substring(idx_headers + separator.length, current.lastIndexOf('\r\n'));
+					let body: T | string = current.substring(idx_headers + separator.length, current.lastIndexOf('\r\n'));
 					let is_json = false;
 
 					tmp = headers['content-type'];
 					if (tmp && !!~tmp.indexOf('application/json')) {
 						try {
-							body = JSON.parse(body);
+							body = JSON.parse(body) as T;
 							is_json = true;
 						} catch (_) {
 						}
 					}
 
-					// @ts-ignore
-					yield { headers, body, json: is_json };
+					tmp = { headers, body, json: is_json };
+					is_eager ? yield tmp : payloads.push(tmp);
 
-					if (next.substring(0, 2) === '--') break outer;
+					// hit a tail boundary, break
+					if (next.substring(0, 2) === '--') {
+						if (payloads.length) {
+							yield payloads;
+						}
+						break outer;
+					}
```

Comment:
@lukeed  do you know of a nicer/better way to do this? This before this commit would not yield the last chunk's payload, seeing as we collect into this array, and "break outer" meaning the yield on line 89 wouldn't run. So this check has to run twice?

Reply:
You could maybe move `payloads` outside the try/finally, and then yield once inside the finally. You'd just have to reset/clear `payloads` as needed